src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -1215,17 +1215,19 @@ private void WriteStartDeltaResourceSetImplementation(ODataDeltaResourceSet delt
 
             this.InterceptException(() =>
             {
-                // Verify query count
-                if (deltaResourceSet.Count.HasValue)

```

Comment:
>deltaResourceSet.Count.HasValue [](start = 20, length = 31)

We don't check the Count for ever? Or accidently remove it?

Reply:
Yes; there didn't seem to be any reason to prohibit writing the count in a request payload, so I removed check entirely. While it doesn't make sense to write nextpagelink or deltalink, I could see services using the count to help process the request.

---
In reply to: [256640820](https://github.com/OData/odata.net/pull/1404#discussion_r256640820) [](ancestors = 256640820)

Reply:
Ok. 

---
In reply to: [256641627](https://github.com/OData/odata.net/pull/1404#discussion_r256641627) [](ancestors = 256641627,256640820)